specs/trip_dispatch_spec.rb

```diff
@@ -111,4 +107,60 @@
       expect(passenger.trips).must_include trip
     end
   end
-end
+
+  describe "In-progress trip" do
+    before do
+      @dispatcher = RideShare::TripDispatcher.new(USER_TEST_FILE,
+        TRIP_TEST_FILE, DRIVER_TEST_FILE)
+      end
+      it "Add's an in-progress trip correctly driver, passenger & dispatched trips" do
+        trip =@dispatcher.request_trip(2)
+        expect(trip).must_be_instance_of RideShare::Trip
+        expect(trip.id).must_equal @dispatcher.trips.length
+        expect(trip.end_time).must_be_nil
+        expect(trip.rating).must_be_nil
+        expect(trip.cost).must_be_nil
+        expect(trip.driver).must_be_instance_of RideShare::Driver
+        expect(trip.passenger).must_be_kind_of RideShare::User
+        expect(trip.driver.status).must_equal :UNAVAILABLE
```

Comment:
There are many more things you should be testing for the `request_trip` method:

- Does the trip reference the correct passenger and driver?
- Was the trip added to the trip lists of the driver and passenger?
- What if there is no available driver?

The last is an important edge case, and is exactly the sort of situation where software could do something unexpected. You might set up the test as follows:

```ruby
it 'raises an error if no driver is available' do
  dispatcher = RideShare::TripDispatcher.new
  dispatcher.drivers.each do |driver|
    # This may require an attr_writer in Driver
    driver.status = :UNAVAILABLE
  end
  proc {
    dispatcher.request_trip(2)
  }.must_raise StandardError
end
```